src/ui/components/search/searchcomponent.js

```diff
@@ -280,6 +284,32 @@ export default class SearchComponent extends Component {
     if (!this._isTwin) {
       this.initQueryUpdateListener();
     }
+
+    /**
+     * If true, a loading indicator will be displayed
+     * @type {boolean}
+     */
+    this._showLoadingIndicator = (config.loadingIndicator && config.loadingIndicator.display) || false;
+
+    /**
+     * Custom icon url for loading indicator
+     * @type {string}
+     */
+    this.loadIconUrl = (config.loadingIndicator && config.loadingIndicator.iconUrl) || null;
+
+    if (this._showLoadingIndicator) {
+      this.core.storage.registerListener({
+        eventType: 'update',
+        storageKey: this._verticalKey ? StorageKeys.VERTICAL_RESULTS : StorageKeys.UNIVERSAL_RESULTS,
+        callback: results => {
+          results.searchState === SearchStates.SEARCH_LOADING
+            ? this.animateIconToLoading()
+            : this.animateIconToDoneLoading();
+        }
+      });
+    }
+
+    this.onMouseUp = this.oneTimeMouseUpListener.bind(this);
```

Comment:
What's this used for?

Reply:
We set iconIsFrozen to true/false in 'mouseup' and 'mousedown' through listeners when user click on elements with '.js-yext-submit' or '.js-yxt-SearchBar-clear'. However, if user somehow move their mouse or lift their finger on mobile that's not on those elements, this will not get trigger and the icons will stay frozen. I updated it to listen on any clicks on 'document' but this may get trigger a lot and we don't need to receive/listen on all the 'mouseup' events. So I made a 'mouseup' listener that get setup when user press down and track 'mouseup' event only once.

Reply:
Does that mean it that the icon can still get frozen if someone lifts their finger off of those elements a second or third time?

Reply:
every time they press down, it setup another oneTimeMouseUpListener!